### DISCUSSION TITLE
Current date as object description

### DISCUSSION THREAD

**Discussion by vaisoliviu** [codefori/vscode-ibmi#2260] [Q&A]
Hello,

I would like to add the current date as the object's description when compiling a program or file.
I was thinking about using an environment variable. Is it possible, and if yes, could u please help me with the variable definition? 

Regards.

**Comment by chrjorgensen** [comment]
@vaisoliviu There is no built-in variable containing the current date... 😞 ...but you can use SQL to change the text and SQL has the current date. 😃 

Your command will depend on what type of source you're compiling from. Assuming your source is in a traditional source member, you could add this command to your compile action (as a command separated from the CRT... command by a newline - you can have more than one command in an action):

```
RUNSQL SQL( 'call QSYS2.QCMDEXC( ''CHGOBJD &OPENLIB/&OPENMBR *PGM TEXT('''''' concat char( current_date ) concat '''''' )'' )' ) COMMIT(*NONE)
```

The same goes for other source member create-commands - and for streamfile sources, you use the variables available streamfile variables:

```
RUNSQL SQL( 'call QSYS2.QCMDEXC( ''CHGOBJD &BUILDLIB/&NAME *PGM TEXT('''''' concat char( current_date ) concat '''''' )'' )' ) COMMIT(*NONE)
```

Or if you prefer, you can create new actions with these commands and run them independently from the compile action.

EDIT: The `RUNSQL` command must of course be in an ILE type action - but you probably already knew that... 😄

**Reply by vaisoliviu** [reply]
@chrjorgensen Thank u for the quick answer. I will use a local variable that I will update manually each day 😄.
